mlir/tools/mlir-miopen-lib/CMakeLists.txt

```diff
@@ -86,53 +86,63 @@ if(BUILD_FAT_LIBROCKCOMPILER)
-  # /a/dummy/lib/cmake/<package>/<package>Targets.cmake
+  # cmake --install . --component ${component_name} --prefix /a/dummy
+  # /a/dummy/lib/${package_name}/lib${lib_name}.a
+  # /a/dummy/lib/cmake/${package_name}/${package_name}Targets.cmake
   #   This file is autogenerated by install(EXPORT).
   #   It creates the imported library as a target that can be used by
   #   target_link_libraries()
-  # /a/dummy/lib/cmake/<package>/<package>Targets-CONFIG.cmake
+  #   It also sets the INTERFACE_INCLUDE_DIRECTORIES property of the target
+  # /a/dummy/lib/cmake/${package_name}/${package_name}Targets-CONFIG.cmake
   #   This file is autogenerated by install(EXPORT).
   #   It sets the location of the library as a property of the imported target
-  # /a/dummy/lib/cmake/<package>/<package>Config.cmake
+  # /a/dummy/lib/cmake/${package_name}/${package_name}Config.cmake
   #   This file is generated by configure_package_config_file and is searched
-  #   and loaded by find_package(). It includes <package>Targets.cmake.
-  # /a/dummy/lib/cmake/<package>/<package>ConfigVersion.cmake
+  #   and loaded by find_package(). It includes ${package_name}Targets.cmake.
+  # /a/dummy/lib/cmake/${package_name}/${package_name}ConfigVersion.cmake
   #   This file is autogenerated by write_basic_package_version_file and is used by
   #   find_package to implement a package version matching mechanism.
   #
-  # The only input to this exporting process is cmake/<package>Config.cmake.in,
-  # which is used to auto generate <package>Config.cmake
-  function(export_target target_name component_name)
-    set(export_set ${PACKAGE_NAME}Targets)
-    set(pkg_config_file ${PACKAGE_NAME}Config.cmake)
-    set(pkg_version_file ${PACKAGE_NAME}ConfigVersion.cmake)
-    set(INCLUDE_DIR include/${PACKAGE_NAME}/)
-    set(LIB_CMAKE_DIR lib/cmake/${PACKAGE_NAME}/)
-
-    # Install lib${target_name}.a to ${CMAKE_INSTALL_PREFIX}/lib/<package>/
+  # The only input to this exporting process is cmake/${package_name}Config.cmake.in,
+  # which is used to auto generate ${package_name}Config.cmake
+  function(export_target package_name lib_name component_name)
+    cmake_parse_arguments(ARG "" "" "EXTRA_INCLUDES" ${ARGN})
+    set(export_set ${package_name}Targets)
+    set(pkg_config_file ${package_name}Config.cmake)
+    set(pkg_version_file ${package_name}ConfigVersion.cmake)
+    set(INCLUDE_DIR include/${package_name}/)
+    set(LIB_CMAKE_DIR lib/cmake/${package_name}/)
+
+    # Install lib${lib_name}.a to ${CMAKE_INSTALL_PREFIX}/lib/${package_name}/
     # and group it into the ${export-set}
     # The INTERFACE_INCLUDE_DIRECTORIES will be set as the property of the exported library
     # so that when comsumer libraries (MIOpen) link in lib${LIBRARY_NAME}, ${INCLUDE_DIR} is
     # automatically propagated to the consumer library's include_dirs
-    install(TARGETS ${target_name}
+    set(EXTRA_INCLUDES "")
+    if(ARG_EXTRA_INCLUDES)
+      set(EXTRA_INCLUDES ${ARG_EXTRA_INCLUDES})
```

Comment:
Question: Is it the core change here, that MIGraphX needs to add `-DARG_EXTRA_INCLUDES`?

Reply:
Might that be set by default?

Reply:
... ah, that's for allowing the user to specify even more includes, I think

Reply:
@krzysz00 By default, the `include/<packagename>` dir is added. Other dirs need to be explicitly specified.

Reply:
Shouldn't those always be installed as part of our package?

That is, wouldn't it make more sense to update `mlir-miopen-lib` to include those migraphx headers?

Reply:
> Shouldn't those always be installed as part of our package?

I think I can see why this decision make sense: MIGraphX and MIOpen required headers are roughly independent. By adding this `extra includes` flag we excluded the migraphX header to be installed into systems that were targeting for MIOpen.